src/languages/clojure/index.ts

```diff
@@ -0,0 +1,196 @@
+  flipBinaryOps,
+  flipped,
+  mapBackwardsIndexToForwards,
+  mapOps,
+  mapOpsTo,
+  removeImplicitConversions,
+} from "../../plugins/ops";
+import {
+  builtin,
+  conditional,
+  functionCall as func,
+  int,
+  op,
+  text,
+  isInt,
+  intToDecOpOrText,
+} from "../../IR";
+import {
+  golfLastPrint,
+  golfLastPrintInt,
+  putcToPrintChar,
+} from "../../plugins/print";
+import { usePrimaryTextOps } from "../../plugins/textOps";
+import { golfStringListLiteral, listOpsToTextOps } from "../../plugins/static";
+import {
+  applyDeMorgans,
+  bitnotPlugins,
+  equalityToInequality,
+  lowBitsPlugins,
+  pickAnyInt,
+  truncatingOpsPlugins,
+} from "../../plugins/arithmetic";
+import { forArgvToForEach } from "../../plugins/loops";
+import { alias, renameIdents } from "../../plugins/idents";
+import { assertInt64 } from "../../plugins/types";
+import { implicitlyConvertConcatArg } from "./plugins";
+
+const clojureLanguage: Language = {
+  name: "Clojure",
+  extension: "clj",
+  emitter: new ClojureEmitter(),
+  phases: [
+    required(arraysToLists, putcToPrintChar, usePrimaryTextOps("codepoint")),
+    simplegolf(golfLastPrint(false), golfLastPrintInt(false)),
+    search(
+      flipBinaryOps,
+      golfStringListLiteral(false),
+      listOpsToTextOps(),
+      equalityToInequality,
+      ...bitnotPlugins,
+      ...lowBitsPlugins,
+      applyDeMorgans,
+    ),
+
+    required(
+      pickAnyInt,
+      forArgvToForEach,
+      mapOps({
+        right_align: (a, b) =>
+          func(
+            "format",
+            op["concat[Text]"](text("%"), intToDecOpOrText(b), text("s")),
+            a,
+          ),
+        int_to_hex_aligned: (a, b) =>
+          func(
+            "format",
+            op["concat[Text]"](text("%0"), intToDecOpOrText(b), text("x")),
+            a,
+          ),
+        int_to_Hex_aligned: (a, b) =>
+          func(
+            "format",
+            op["concat[Text]"](text("%0"), intToDecOpOrText(b), text("X")),
+            a,
+          ),
+      }),
+    ),
+    simplegolf(implicitlyConvertConcatArg),
+    required(
+      mapOps({
+        argv: () => builtin("*command-line-args*"),
+
+        // TODO: vec not necessary if already a vec
+        append: (a, b) => func("conj", a, b),
+
+        "at[argv]": (a) => op["at[List]"](builtin("*command-line-args*"), a),
+        "at[codepoint]": (a, b) => func("str", func("nth", a, b)),
+        "contains[Text]": (a, b) => func("clojure.string/includes?", a, b),
+        "contains[Table]": (a, b) => func("contains?", a, b),
+        "ord[codepoint]": (a) => op["ord_at[codepoint]"](a, int(0n)),
+        pow: (a, b) => func("int", func("Math/pow", a, b)),
+      }),
+      mapOps({
+        "at_back[List]": (a, b) =>
+          isInt(-1n)(b) ? func("last", a) : undefined,
+      }),
+      mapBackwardsIndexToForwards({
+        "at_back[Ascii]": "size[Ascii]",
+        "at_back[codepoint]": "size[codepoint]",
+        "at_back[List]": "size[List]",
```

Comment:
`at_back[List]` is already mapped using the previous plugin

Reply:
But it returns `undefined` unless it's -1, so this has to be implemented too.

Reply:
> But it returns `undefined` unless it's -1, so this has to be implemented too.

yes, mb